docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/TimeSeries/DetectAnomalyBySrCnn.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.ML;
+using Microsoft.ML.Data;
+using Microsoft.ML.Transforms.TimeSeries;
+
+namespace Samples.Dynamic
+{
+    public static class DetectAnomalyBySrCnn
+    {
+        // This example creates a time series (list of Data with the i-th element corresponding to the i-th time slot). 
+        // The estimator is applied then to identify spiking points in the series.
+        public static void Example()
+        {
+            // Create a new ML context, for ML.NET operations. It can be used for exception tracking and logging, 
+            // as well as the source of randomness.
+            var ml = new MLContext();
+
+            // Generate sample series data with a spike
+            const int Size = 10;
+            var data = new List<TimeSeriesData>(Size + 1)
+            {
+                new TimeSeriesData(5),
+                new TimeSeriesData(5),
+                new TimeSeriesData(5),
+                new TimeSeriesData(5),
+                new TimeSeriesData(5),
+
+                // This is a spike.
+                new TimeSeriesData(10),
+
+                new TimeSeriesData(5),
+                new TimeSeriesData(5),
+                new TimeSeriesData(5),
+                new TimeSeriesData(5),
+                new TimeSeriesData(5),
+            };
+
+            // Convert data to IDataView.
+            var dataView = ml.Data.LoadFromEnumerable(data);
+
+            // Setup IidSpikeDetector arguments
+            string outputColumnName = nameof(SrCnnAnomalyDetection.Prediction);
+            string inputColumnName = nameof(TimeSeriesData.Value);
+
+            // The transformed model.
+            ITransformer model = ml.Transforms.DetectAnomalyBySrCnn(outputColumnName, inputColumnName, 64, 5, 5, 3, 21, 0.25).Fit(dataView);
+
+            // Create a time series prediction engine from the model.
+            var engine = model.CreateTimeSeriesPredictionFunction<TimeSeriesData, SrCnnAnomalyDetection>(ml);
+
+            Console.WriteLine($"{outputColumnName} column obtained post-transformation.");
+
+            // Create non-anomalous data and check for anomaly.
+            for (int index = 0; index < 100; index++)
+            {
+                // Anomaly spike detection.
+                PrintPrediction(5, engine.Predict(new TimeSeriesData(5)));
```

Comment:
Please add the expected output lines to this sample.

Reply:
Done.